move_base/src/move_base.cpp

```diff
@@ -1117,8 +1123,7 @@ namespace move_base {
           state_ = CLEARING;
           recovery_trigger_ = OSCILLATION_R;
         }
-        {
-         boost::unique_lock<costmap_2d::Costmap2D::mutex_t> lock(*(controller_costmap_ros_->getCostmap()->getMutex()));
+        boost::unique_lock<costmap_2d::Costmap2D::mutex_t> lock(*(controller_costmap_ros_->getCostmap()->getMutex()));
```

Comment:
unique_lock is a scoped lock.  It unlocks when it goes out of scope. Those extra braces provide a scope for it.